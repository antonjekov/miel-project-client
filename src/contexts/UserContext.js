import React from "react";

const UserContext = React.createContext({name: "Anton"});

export default UserContext