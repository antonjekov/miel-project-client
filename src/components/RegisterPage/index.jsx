import { Form, Button, Col, Spinner } from 'react-bootstrap';
import React, { useState } from 'react';
import userService from "../../services/user_service";
import userSchema from "../../schemas/userSchema";
import { useFormik } from "formik";
import { Redirect,Link } from "react-router-dom";
import { useAuth } from "../../contexts/Auth";
import FormContentWrapper from "../FormContentWrapper"
import PasswordInput from "../PasswordInput";
import RepeatPasswordInput from "../RepeatPasswordInput";
import EmailInput from "../EmailInput";
import NameInput from "../NameInput";


function Register(props) {
    const { setUserInfo } = useAuth();
    const [isLoading, setIsLoading] = useState(false);
    const [loginSuccess, setLoginSuccess] = useState(false);

    //Formik integration hook
    const { handleSubmit, handleChange, errors, values } = useFormik({
        initialValues: {
            username: '',
            password: '',
            repassword: '',
            name: '',
        },
        validationSchema: userSchema,
        onSubmit(values, actions) {
            setIsLoading(true);
            userService.register(values)
                .then(async res => {
                    if (!res.ok) {
                        setIsLoading(false);
                        const errorsObject = await res.json()
                        actions.setErrors(errorsObject)
                        return
                    }
                    const userInfo = await res.json()
                    setUserInfo(userInfo)
                    setIsLoading(false);
                    setLoginSuccess(true);
                });
        }

    });

    return (loginSuccess ? <Redirect to='/' /> :
        <Col md={{ offset: 2, span: 8 }}>
            <FormContentWrapper title='Register'>
                <Form onSubmit={handleSubmit}>
                    <Form.Row>
                        <Col>
                            <NameInput values={values} errors={errors} handleChange={handleChange} />
                        </Col>
                        <Col>
                            <EmailInput values={values} errors={errors} handleChange={handleChange} />
                        </Col>
                    </Form.Row>

                    <Form.Row>
                        <Col>
                            <PasswordInput values={values} errors={errors} handleChange={handleChange} />
                        </Col>
                        <Col>
                            <RepeatPasswordInput values={values} errors={errors} handleChange={handleChange} />
                        </Col>
                    </Form.Row>
                    {isLoading ?
                        <Spinner animation="border" variant="warning" /> :
                        <Button variant="warning" type="submit" >Register</Button>}
                    <Form.Text >
                        Already have an account? <Link to="/login" ><b>Login</b></Link>
                    </Form.Text>
                </Form>
            </FormContentWrapper>
        </Col>

    );
}

export default Register