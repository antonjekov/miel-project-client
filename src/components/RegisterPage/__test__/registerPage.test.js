import React from "react";
import ReactDOM from "react-dom"
import {render, cleanup, screen, fireEvent} from "@testing-library/react"
import renderer from "react-test-renderer"
import RegisterPage from "../"
import { useAuth, AuthContext } from  "../../../contexts/Auth";
import { BrowserRouter, Switch, Route } from "react-router-dom";

afterEach(cleanup)

const setUserInfo =()=>{}
test("inputs update on change", () =>{
    const {queryByPlaceholderText, queryByText} = render(<BrowserRouter><AuthContext.Provider value={{ setUserInfo}}><RegisterPage/></AuthContext.Provider></BrowserRouter>)
    const nameInput = queryByPlaceholderText('Enter name')
    fireEvent.change(nameInput, {target:{value: 'Anton'}})
    expect(nameInput.value).toBe('Anton')
    const emailInput = queryByPlaceholderText('Enter email')
    fireEvent.change(emailInput,{target:{value: 'email'}})
    expect(emailInput.value).toBe('email')
    const passwordInput = queryByPlaceholderText('Enter your password')
    fireEvent.change(passwordInput,{target:{value: '123456'}})
    expect(passwordInput.value).toBe('123456')
    const repasswordInput = queryByPlaceholderText('Repeat Password')
    fireEvent.change(repasswordInput,{target:{value: '123456'}})
    expect(repasswordInput.value).toBe('123456')
  })

  //Snapshot tests
it('matches snapshot', ()=>{
  const tree = renderer.create(
    <BrowserRouter><AuthContext.Provider value={{setUserInfo}}>
          <RegisterPage/>
      </AuthContext.Provider></BrowserRouter>
 ).toJSON();

  expect(tree).toMatchSnapshot();
})