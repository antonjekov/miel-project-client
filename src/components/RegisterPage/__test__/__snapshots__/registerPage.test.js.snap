// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="col-md-8 offset-md-2"
>
  <div
    className="FormWrapper"
  >
    <h1>
      Register
    </h1>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <div
          className="col"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Name
            </label>
            <input
              className="form-control form-control"
              name="name"
              onChange={[Function]}
              placeholder="Enter name"
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            />
            <div
              className="valid-feedback"
            >
              Look's good
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Email address
            </label>
            <input
              className="form-control form-control"
              name="username"
              onChange={[Function]}
              placeholder="Enter email"
              type="email"
              value=""
            />
            <div
              className="invalid-feedback"
            />
            <div
              className="valid-feedback"
            >
              Look's good
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="col"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Password
            </label>
            <input
              className="form-control form-control"
              name="password"
              onChange={[Function]}
              placeholder="Enter your password"
              type="password"
              value=""
            />
            <div
              className="invalid-feedback"
            />
            <div
              className="valid-feedback"
            >
              Look's good
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Repeat Password
            </label>
            <input
              className="form-control form-control"
              name="repassword"
              onChange={[Function]}
              placeholder="Repeat Password"
              type="password"
              value=""
            />
            <div
              className="invalid-feedback"
            />
            <div
              className="valid-feedback"
            >
              Look's good
            </div>
          </div>
        </div>
      </div>
      <button
        className="btn btn-warning"
        disabled={false}
        type="submit"
      >
        Register
      </button>
      <small
        className="form-text"
      >
        Already have an account? 
        <a
          href="/login"
        >
          <b>
            Login
          </b>
        </a>
      </small>
    </form>
  </div>
</div>
`;
